pkg/util/mount/mount_linux.go

```diff
@@ -430,3 +430,43 @@ func (mounter *SafeFormatAndMount) getDiskFormat(disk string) (string, error) {
 	// and MD RAID are reported as FSTYPE and caught above).
 	return "unknown data, probably partitions", nil
 }
+
+// IsSamefsGroup is called only for requests to mount an already mounted
+// volume. It checks if fsGroup of new mount request is the same or not.
+// It return false if it not the same.
+func IsSamefsGroup(dir string, fsGroup *int64) (bool, error) {
+        glog.V(4).Infof("######### IsSamefsGroup path %s ###########",dir)
+        s := &syscall.Stat_t{}
+        if err := syscall.Stat(dir, s); err != nil {
+                glog.V(4).Infof("######### IsSamefsGroup failed to get stat for %s (%v)###########",dir,err)
+                return false, err
+        }
+        glog.V(4).Infof("######### IsSameGroup  Old fsGroup - %d New fsGroup - %d ###########",s.Gid, fsGroup)
+        if int(s.Gid) == int(*fsGroup) {
+          glog.V(4).Infof("######### fsGroups same ###########")
+          return true, nil
+        }
+        glog.V(4).Infof("######### fsGrous different ###########")
+        return false, nil
+}
+
+// IsAlreadyMounted is called to check if a requested volume has already been
+// mounted. It returns nil if no previous mount was found, otherwise it returns
+// the path to existing mount.
+func IsAlreadyMounted(dir string) (string, error) {
+   r := ""
+   mp, err := listProcMounts(procMountsPath)
+   if err != nil {
+     return r, err
+   }
+   fullPath := strings.Split(dir, "/")
+   pvc := fullPath[len(fullPath)-1]
```

Comment:
I was originally thinking that you could take the global mount path and see if that is bind-mounted multiple times.